frontend/src/AccountCreation/SignUpProfessor.tsx

```diff
@@ -4,26 +4,104 @@ import Stack from "react-bootstrap/Stack";
 import { Button } from "react-bootstrap";
 import Form from "react-bootstrap/Form";
 import Alert from "react-bootstrap/Alert";
-import React, { useState } from "react";
+import React, { FormEvent, useState } from "react";
 
 const SignUpProfessor = () => {
   const nav = useNavigate();
   const [error, setError] = useState("");
 
-  const submit: React.FormEventHandler<HTMLFormElement> = (
-    e: React.FormEvent<HTMLFormElement>
-  ) => {
+  const [name, setName] = useState("");
+  const [uid, setUID] = useState("");
+  const [courseName, setCourseName] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+
+  //check if passwords match
+  function validateForm(
+    name: string,
+    uid: string,
+    course_name: string,
+    password: string,
+    confirm_password: string
+  ) {
+    if (name.length === 0) {
+      setError("Please enter your name");
+      return false;
+    }
+    if (uid.length !== 9 || isNaN(parseInt(uid))) {
+      setError("Please correct your UID");
+      return false;
+    }
+    if (course_name.length === 0) {
+      setError("Please enter a course name");
+      return false;
+    } else if (password !== confirm_password) {
+      setError("Passwords do not match");
+      return false;
+    } else if (password.length < 8) {
+      setError("Password must be at least 8 characters long");
+      return false;
+    }
+    return true;
+  }
```

Comment:
Small note. Form validation can actually be carried out in HTML5 natively (see [here](https://www.sitepoint.com/html5-form-validation/)) to avoid writing quite a few validation functions. If the submitted information isn't what we want on the backend, we can add constraints later.

Reply:
Fixed